plugins/UM3NetworkPrinting/src/UM3OutputDevicePlugin.py

```diff
@@ -370,3 +400,124 @@ def _onServiceChanged(self, zero_conf, service_type, name, state_change):
+            return
+
+    def _onCloudPrintingConfigured(self):
+        if self._start_cloud_flow_message:
+            self._start_cloud_flow_message.hide()
+            self._start_cloud_flow_message = None
+        
+        # Show the successful pop-up
+        if not self._start_cloud_flow_message:
+            self._cloud_flow_complete_message = Message(
+                i18n_catalog.i18nc("@info:status", "You can now send and monitor print jobs from anywhere using your Ultimaker account."),
+                30, # Lifetime
+                True, # Dismissable?
+                None, # Progress
+                "", # Title
+                None, # Parent
+                True, # Use inactivity timer
+                "../../../../../Cura/plugins/UM3NetworkPrinting/resources/svg/cloud-flow-completed.svg", # Image souce
+                i18n_catalog.i18nc("@info:status", "Connected!") # image caption
+            )
+            self._cloud_flow_complete_message.addAction("", i18n_catalog.i18nc("@action", "Review your connection"), "", "", 1) # TODO: Icon
+            self._cloud_flow_complete_message.actionTriggered.connect(self._onReviewCloudConnection)
+            self._cloud_flow_complete_message.show()
+
+            # Set the machine's cloud flow as complete so we don't ask the user again and again for cloud connected printers
+            active_machine = self._application.getMachineManager().activeMachine
+            if active_machine:
+                active_machine.setMetaDataEntry("cloud_flow_complete", True)
+            return
+
+    def _onDontAskMeAgain(self, message):
+        active_machine = self._application.getMachineManager().activeMachine
+        if active_machine:
+            active_machine.setMetaDataEntry("show_cloud_message", False)
+            Logger.log("d", "Will not ask the user again to cloud connect for current printer.")
+        return
+
+    def _onCloudFlowStarted(self, message, action):
+        address = self._application.getMachineManager().activeMachineAddress
+        if address:
+            QDesktopServices.openUrl(QUrl("http://" + address + "/cloud_connect"))
+            if self._start_cloud_flow_message:
+                self._start_cloud_flow_message.hide()
+                self._start_cloud_flow_message = None
+        return
+    
+    def _onReviewCloudConnection(self, message, action):
+        address = self._application.getMachineManager().activeMachineAddress
+        if address:
+            QDesktopServices.openUrl(QUrl("http://" + address + "/settings"))
+        return
+
+    def _onMachineSwitched(self):
+        if self._start_cloud_flow_message is not None:
+            self._start_cloud_flow_message.hide()
+            self._start_cloud_flow_message = None
+        if self._cloud_flow_complete_message is not None:
+            self._cloud_flow_complete_message.hide()
+            self._cloud_flow_complete_message = None
+
+        self.checkCloudFlowIsPossible()
```

Comment:
This whole file has a deficiency in Python type hints, mostly in the method defs.

Reply:
I'll add them in.